File: app/src/main/java/ru/SnowVolf/convertx/ui/Interfacer.java
package ru.SnowVolf.convertx.ui;

import android.app.Activity;
import android.content.Context;
import android.content.res.Resources;
import android.content.res.TypedArray;

import ru.SnowVolf.convertx.App;
import ru.SnowVolf.convertx.R;
import ru.SnowVolf.convertx.settings.SettingsFragment;

/**
 * Created by Snow Volf on 29.06.2017, 20:48
 */

public abstract class Interfacer {
    public enum Theme{
        LIGHT,
        DARK,
        PURE_DARK
    }

    public static void applyTheme(Activity ctx){
        int theme;
        switch (Theme.values()[SettingsFragment.getThemeIndex(ctx)]){
            case LIGHT:
                theme = R.style.MaterialDrawerTheme_Light_DarkToolbar;
                break;
            case DARK:
                theme = R.style.MaterialDrawerTheme_TranslucentStatus;
                break;
            case PURE_DARK:
                theme = R.style.MaterialDrawerTheme_TranslucentStatusBlack;
                break;
            default:
                theme = R.style.MaterialDrawerTheme_Light_DarkToolbar;
                break;
        }
        ctx.setTheme(theme);
    }

    public static int getThemedResourceId(Context ctx, int attrId){
        return getThemedResourceId(ctx.getTheme(), attrId);
    }

    public static int getThemedResourceId(Resources.Theme theme, int attrId){
        TypedArray a = theme.obtainStyledAttributes(new int[]{attrId});
        int result = a.getResourceId(0, -1);
        a.recycle();
        return result;
    }

    public static int getThemedColor(Context ctx, int attrId){
        return getThemedColor(ctx.getTheme(), attrId);
    }

    public static int getThemedColor(Resources.Theme theme, int attrId){
        TypedArray a = theme.obtainStyledAttributes(new int[]{attrId});
        int result = a.getResourceId(0, -1);
        a.recycle();
        return result;
    }

    public static void setThemeIndex(int index){
        //noinspection ConstantConditions
        App.Companion.getInstance().getPrefs().edit().putInt("ITheme.Theme", index).apply();
    }

}

File: app/src/main/java/ru/svolf/convertx/ui/Interfacer.kt
package ru.svolf.convertx.ui

import ru.svolf.convertx.App.Companion.instance
import android.app.Activity
import android.content.Context
import android.content.res.Resources
import ru.svolf.convertx.settings.SettingsFragment
import ru.svolf.convertx.R

/**
 * Created by Snow Volf on 29.06.2017, 20:48
 */
object Interfacer {
    fun applyTheme(ctx: Activity) {
        val theme: Int
        theme = when (Theme.values()[SettingsFragment.getThemeIndex(ctx)]) {
            Theme.LIGHT -> R.style.MaterialDrawerTheme_Light_DarkToolbar
            Theme.DARK -> R.style.MaterialDrawerTheme_TranslucentStatusBlack
            else -> R.style.MaterialDrawerTheme_Light_DarkToolbar
        }
        ctx.setTheme(theme)
    }

    fun getThemedResourceId(ctx: Context, attrId: Int): Int {
        return getThemedResourceId(ctx.theme, attrId)
    }

    fun getThemedResourceId(theme: Resources.Theme, attrId: Int): Int {
        val a = theme.obtainStyledAttributes(intArrayOf(attrId))
        val result = a.getResourceId(0, -1)
        a.recycle()
        return result
    }

    fun getThemedColor(ctx: Context, attrId: Int): Int {
        return getThemedColor(ctx.theme, attrId)
    }

    fun getThemedColor(theme: Resources.Theme, attrId: Int): Int {
        val a = theme.obtainStyledAttributes(intArrayOf(attrId))
        val result = a.getResourceId(0, -1)
        a.recycle()
        return result
    }

    fun setThemeIndex(index: Int) {
        instance!!.prefs.edit().putInt("ITheme.Theme", index).apply()
    }

    enum class Theme {
        LIGHT, DARK
    }
}